### ISSUE TITLE
Incorrect type when `Enum` member references other `Enum` member in initalization

### ISSUE THREAD

**Issue by StefanBRas** [microsoft/pyright#7763]
**Describe the bug**
If I reference another member at initialization time, pyright says that i have to say `val.value` but at runtime it's just `val`

**Code or Screenshots**
This is working code that pyright rejects:
```python
from enum import Enum

class A(Enum):
    a = 1
    b = a + 2  # Operator "+" not supported for types "A" and "Literal[2]"

print(A.b.value) # prints 3
```
This is what pyright wants:
```python
from enum import Enum

class A(Enum):
    a = 1
    b = a.value + 2

```
But this fails with
```bash
Traceback (most recent call last):
  File "/Users/sras/dev/test_pyright/src/test_pyright/__init__.py", line 3, in <module>
    class A(Enum):
  File "/Users/sras/dev/test_pyright/src/test_pyright/__init__.py", line 5, in A
    b = a.value + 2
        ^^^^^^^
AttributeError: 'int' object has no attribute 'value'

```

**VS Code extension or command-line**
Pyright v1.1.360, python v.3.12.2. Tested both as lsp in neovim and as pip installed cli tool.

**Comment by erictraut** [comment]
The behavior here is very difficult for a static type checker to model because the type of `a` is modified by the Enum metaclass, which performs a lot of "magic". The type of `a` is `int` when accessed within the class body but then changes to `Literal[a]` when accessed outside of the class body. 

I'll investigate to see if there's a way I can support this behavior in a straightforward way, but this may be something I close as "won't fix".

It appears that mypy does support this behavior.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.363, which I just published.
